go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+
+	tests := []struct {
+		desc                     string
+		inputLink                netlink.Link
+		inputNewAddr             *net.IPNet
+		errExp                   bool
+		onRetArgsNetLinkLibOpers []onCallReturnArgs
+		onRetArgsLinkIfaceOpers  []onCallReturnArgs
+	}{
+		// TODO: Code may need to be updated to check that link is not-nil to avoid panic
+		/*{
+			desc: "link is nil",
+			inputLink: nil,
+			inputNewAddr: nil,
+			errExp: true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{nil, fmt.Errorf("mock error")}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{},[]interface{}{&netlink.LinkAttrs{Name: "testIfaceName",}}},
+			},
+		},*/
+		{
+			desc:         "setting <nil> address on link errors out",
+			inputLink:    mockLink,
+			inputNewAddr: nil,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{fmt.Errorf("mock error")}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName"}}},
+			},
+		},
+		{
+			desc:         "test code path where error is returned when attempting to set new address on link",
+			inputLink:    mockLink,
+			inputNewAddr: ovntest.MustParseIPNet("192.168.1.15/24"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{fmt.Errorf("mock error")}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName"}}},
+			},
+		},
+		{
+			desc:         "setting new address on link succeeds",
+			inputLink:    mockLink,
+			inputNewAddr: ovntest.MustParseIPNet("192.168.1.15/24"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"AddrAdd", []string{"*mocks.Link", "*netlink.Addr"}, []interface{}{nil}},
+			},
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			for _, item := range tc.onRetArgsLinkIfaceOpers {
+				call := mockLink.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+			err := LinkAddrAdd(tc.inputLink, tc.inputNewAddr)
+			t.Log(err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
```

Comment:
L396: are you missing the`else` part here

Reply:
Yes, it makes sense to add else that asserts err  is not nil.